pkgs/development/rocm-modules/5/rocprofiler/default.nix

```diff
@@ -79,16 +93,31 @@ stdenv.mkDerivation (finalAttrs: {
   postPatch = ''
     patchShebangs .
 
-    # Cannot find ROCm device library, pointless
-    substituteInPlace CMakeLists.txt \
-      --replace "add_subdirectory(tests-v2)" "" \
-      --replace "add_subdirectory(samples)" ""
+    # Fixup bad paths
+    substituteInPlace bin/rpl_run.sh \
+      --replace "ROOT_DIR=\$(dirname \$BIN_DIR)" "ROOT_DIR=${rocmtoolkit-merged}" \
+      --replace "RPL_PATH=\$ROOT_DIR/lib" "RPL_PATH=\$(dirname \$BIN_DIR)/lib" \
+      --replace "PROF_BIN_DIR=\$ROOT_DIR" "PROF_BIN_DIR=\$(dirname \$BIN_DIR)" \
+      --replace "ROCP_PACKAGE_DIR=\$ROOT_DIR" "ROCP_PACKAGE_DIR=\$(dirname \$BIN_DIR)" \
+      --replace "'\$ROOT_DIR'" "'\$(dirname \$BIN_DIR)'" \
+      --replace "/bin/ls" "ls"
+
+    substituteInPlace bin/rocprofv2 \
+      --replace "ROCM_DIR=\$(dirname -- \"\$ROCPROFV2_DIR\")" "ROCPROFILER_DIR=\$(dirname -- \"\$ROCPROFV2_DIR\")''\nROCM_DIR=${rocmtoolkit-merged}" \
+      --replace "\$ROCM_DIR/lib/rocprofiler" "\$ROCPROFILER_DIR/lib/rocprofiler" \
+      --replace "\$ROCM_DIR/libexec/rocprofiler" "\$ROCPROFILER_DIR/libexec/rocprofiler"
+
+    substituteInPlace tests-v2/featuretests/profiler/CMakeLists.txt \
+      --replace "--build-id=sha1" "--build-id=sha1 --rocm-path=${clr} --rocm-device-lib-path=${rocm-device-libs}/amdgcn/bitcode"
+
+    substituteInPlace test/CMakeLists.txt \
+      --replace "\''${ROCM_ROOT_DIR}/amdgcn/bitcode" "${rocm-device-libs}/amdgcn/bitcode"
   '';
 
-  postBuild = ''
-    # HSACO aren't being built for some reason
-    substituteInPlace test/cmake_install.cmake \
-      --replace "file(INSTALL DESTINATION \"\''${CMAKE_INSTALL_PREFIX}/share/rocprofiler/tests-v1\" TYPE FILE FILES \"" "message(\""
+  postInstall = ''
+    # ???
```

Comment:
I am not sure what that comment should tell the reader

Reply:
Confusion, on why the two libraries below didn't already have the executable bit.